tests/providers/google/cloud/utils/test_credentials_provider.py

```diff
@@ -327,11 +327,28 @@ def test_get_credentials_and_project_id_with_mutually_exclusive_configuration(
         'google.oauth2.service_account.Credentials.from_service_account_file',
     )
     def test_disable_logging(self, mock_default, mock_info, mock_file):
-        # assert not logs
+        """
+        Test disable logging in ``get_credentials_and_project_id``.
+
+        Due to selected limitation:
+        - Unable mixin pytest autouse-fixture `caplog` with `unittest.TestCase`
+        - `unittest.TestCase.assertNoLogs` available only in Python 3.10+
+
+        We use some workarounds for filtering specific logger and raise error with these records.
+        """
+        logger_name = "airflow.providers.google.cloud.utils.credentials_provider._CredentialProvider"
+
```

Comment:
small nitty suggestion.... move this line below function def so it's closer to where it's used / easier to see and find

Reply:
Good idea. Initially I keep it outside for use it with pure `pytest`.